src/storage/index/LookupBaseProcessor.inl

```diff
@@ -33,17 +33,41 @@ cpp2::ErrorCode LookupBaseProcessor<REQ, RESP>::requestCheck(const cpp2::LookupI
 
     // setup yield columns.
     if (req.__isset.return_columns) {
-        const auto& retcols = *req.get_return_columns();
-        yieldCols_ = retcols;
+        yieldCols_ = *req.get_return_columns();
+    }
+
+    for (const auto& ctx : contexts_) {
+        const auto& indexId = ctx.get_index_id();
+        auto needFilter = ctx.__isset.filter && !ctx.get_filter().empty();
+
+        auto index = planContext_->isEdge_
+            ? this->env_->indexMan_->getEdgeIndex(spaceId_, indexId)
+            : this->env_->indexMan_->getTagIndex(spaceId_, indexId);
+        if (!index.ok()) {
+            return cpp2::ErrorCode::E_INDEX_NOT_FOUND;
+        }
+
+        auto fields = index.value()->get_fields();
+
+        // If no filed in fields, only _vid can be returned for tag.
+        // For edge, only _src, _ranking, _dst can be returned
+        if (fields.size() == 0) {
+            //
+            if (contexts_.size() != 1 || yieldCols_.size() != 0 ||
+                needFilter || ctx.get_column_hints().size() != 0) {
+                return cpp2::ErrorCode::E_INVALID_OPERATION;
+            }
+            noProp_ = true;
+        }
```

Comment:
@bright-starry-sky, would you help to review this piece of block? It supposes to make sure that the query looks like `lookup on all tags where tag == xxx`, but the implementation confuses me.

Reply:
> @bright-starry-sky, would you help to review this piece of block? It supposes to make sure that the query looks like `lookup on all tags where tag == xxx`, but the implementation confuses me.

I guess，when` lookup on all tags where tag == xxx` , the IndexQueryContext should be as below :
```
struct IndexQueryContext {
    1: common.IndexID           index_id,        ----> The actual index ID
    // filter is an encoded expression of where clause.
    // Used for secondary filtering from a result set
    2: binary                   filter,                       -----> null
    // There are two types of scan: 1, range scan; 2, match scan (prefix);
    // When the field size of index_id IndexItem is not zero, the columns_hints are not allowed
    //    to be empty, At least one index column must be hit.
    // When the field size of index_id IndexItem is zero, the columns_hints must be empty.
    3: list<IndexColumnHint>    column_hints,      -----> empty
}
```

So just index id needed. right ？ if yes , I think the logic is good for me. A small suggestion is put this check logic outside the for loop, because the contexts_.size() must be one when  `where tag == xxx`.

Reply:
Agreed. Just check first context is more clear.

Reply:
> A small suggestion is put this check logic outside the for loop, because the contexts_.size() must be one when `where tag == xxx`.

I put it here on purpose.
The reason is that if an ordinary lookup query contains an index without fields, an error will be reported.

So here need to check each `IndexQueryContext`.

Reply:
> > A small suggestion is put this check logic outside the for loop, because the contexts_.size() must be one when `where tag == xxx`.
> 
> I put it here on purpose.
> The reason is that if an ordinary lookup query contains an index without fields, an error will be reported.
> 
> So here need to check each `IndexQueryContext`.

Aha, If it were me, I would be inclined to do so as bellow. just my own thoughts

```
if (contexts_.size() == 1 && contexts_[0].column_hints.empty()) {
    ...
} else {
   ...
}
```

Reply:
I got you.
Here  includes the following two meanings:
1) When there is only an index with no fields, meet specific conditions
2) When an ordinary lookup, this special index cannot be included.

So I think every element in contexts_ needs to be checked.